### PR TITLE
Update web viewer and desktop viewer templates to latest versions of measure-tools-react, tree-widget-react, and property-grid-react

### PR DESCRIPTION
**Pull request by Jake-Screen**
- Updating imported versions of measure-tools, tree-widget, and property-grid packages from viewer-components-react in web and desktop templates
- Updated core and appui packages to 4.0.0-dev versions in web viewer template 
- updated Viewer component in web viewer template, adding changesetid and backendConfiguration  props.
-  added MeasurementActionToolbar to `onIModelAppInit` callback to allow popup toolbar from measure-tools-react to show
- Updated desktop template to 4.0.0-dev versions of core and appui packages
- added rpcInterfaces and clientID to DesktopInitializerParams and viewer props

### CODE REVIEW

File: packages/templates/cra-template-web-viewer/template/src/App.tsx [hunk 1]
```diff
@@ -9,6 +9,20 @@ import type { ScreenViewport } from "@itwin/core-frontend";
...
+import {
+  MeasurementActionToolbar,
+  MeasureTools,
+  MeasureToolsUiItemsProvider,
+} from "@itwin/measure-tools-react";
+import {
+  PropertyGridManager,
+  PropertyGridUiItemsProvider,
+} from "@itwin/property-grid-react";
+// import LocalExtension from "@itwin/test-local-extension";
```
Comment (aruniverse) [inline packages/templates/cra-template-web-viewer/template/src/App.tsx:21]: ?

Comment (Jake-Screen) [reply packages/templates/cra-template-web-viewer/template/src/App.tsx:21]: removed

File: packages/templates/cra-template-desktop-viewer/template.json [hunk 2]
```diff
@@ -68,6 +68,11 @@
       "start": "npm run build:backend && run-p \"start:frontend\" \"electron:debug\"",
       "start:frontend": "react-scripts start"
     },
+    "pnpm": {
```
Comment (aruniverse) [inline packages/templates/cra-template-desktop-viewer/template.json:71]: if youre going to add the override, then i would recommend changing the recommended approach in the readme from using npx to pnpm dlx, take a look at docs for that

Comment (Jake-Screen) [reply packages/templates/cra-template-desktop-viewer/template.json:71]: changed readmes

File: packages/templates/cra-template-web-viewer/template/.env [hunk 3]
```diff
@@ -4,6 +4,7 @@ IMJS_AUTH_CLIENT_REDIRECT_URI = ""
 IMJS_AUTH_CLIENT_LOGOUT_URI = ""
 IMJS_AUTH_CLIENT_SCOPES ="imodelaccess:read imodels:read realitydata:read"
 IMJS_AUTH_AUTHORITY="https://ims.bentley.com"
+IMJS_URL_PREFIX=""
```
Comment (aruniverse) [inline packages/templates/cra-template-web-viewer/template/.env:7]: remove, i dont want to expose this to users

File: packages/templates/cra-template-web-viewer/template/src/index.tsx [hunk 4]
```diff
@@ -12,6 +12,8 @@ import App from "./App";
 import { Auth } from "./Auth";
 import * as serviceWorker from "./serviceWorker";
 
+globalThis.IMJS_URL_PREFIX = process.env.IMJS_URL_PREFIX || "";
```
Comment (aruniverse) [inline packages/templates/cra-template-web-viewer/template/src/index.tsx:15]: remove, dont want to expose this to users, they should always point to prod

File: packages/templates/cra-template-web-viewer/template.json [hunk 5]
```diff
@@ -1,58 +1,73 @@
...
+      }
+    },
+    "eslintConfig": {
+      "extends": [
+        "react-app",
+        "react-app/jest"
+      ]
     },
     "browserslist": [
-      "last 4 chrome version",
```
Comment (aruniverse) [inline packages/templates/cra-template-web-viewer/template.json:48]: revert these changes

File: packages/templates/cra-template-web-viewer/template.json [hunk 6]
```diff
@@ -1,58 +1,73 @@
...
-      "extends": ["react-app", "react-app/jest"]
-    },
     "scripts": {
       "start": "react-scripts --max_old_space_size=4096 start",
       "build": "react-scripts --max_old_space_size=4096 build",
-      "test": "react-scripts --max_old_space_size=4096 test"
+      "test": "react-scripts --max_old_space_size=4096 test",
+      "eject": "react-scripts eject"
+    },
+    "resolutions": {
```
Comment (aruniverse) [inline packages/templates/cra-template-web-viewer/template.json:51]: this field isnt applicable if youre using pnpm

File: packages/templates/cra-template-web-viewer/template.json [hunk 7]
```diff
@@ -1,58 +1,73 @@
 {
   "package": {
     "dependencies": {
-      "@bentley/icons-generic": "^1.0.13",
-      "@bentley/react-scripts": "^5.0.4",
-      "@itwin/core-telemetry": "^3.7.0",
-      "@itwin/appui-abstract": "^3.7.0",
+      "@bentley/icons-generic": "1.0.13",
```
Comment (aruniverse) [inline packages/templates/cra-template-web-viewer/template.json:4]: why are deps being pinned here?, they should be using ^

File: packages/templates/cra-template-web-viewer/template/.env (hunk 3 above)
Comment (Jake-Screen) [reply packages/templates/cra-template-web-viewer/template/.env]: removed

File: packages/templates/cra-template-web-viewer/template/src/index.tsx (hunk 4 above)
Comment (Jake-Screen) [reply packages/templates/cra-template-web-viewer/template/src/index.tsx]: removed

File: packages/templates/cra-template-web-viewer/template.json (hunk 6 above)
Comment (Jake-Screen) [reply packages/templates/cra-template-web-viewer/template.json]: removed field

File: packages/templates/cra-template-web-viewer/template.json (hunk 7 above)
Comment (Jake-Screen) [reply packages/templates/cra-template-web-viewer/template.json]: unpinned

File: packages/templates/cra-template-web-viewer/template/src/index.tsx [hunk 8]
```diff
@@ -29,9 +29,9 @@ if (!process.env.IMJS_AUTH_CLIENT_REDIRECT_URI) {
 }
 
 Auth.initialize({
-  scope: process.env.IMJS_AUTH_CLIENT_SCOPES,
-  clientId: process.env.IMJS_AUTH_CLIENT_CLIENT_ID,
-  redirectUri: process.env.IMJS_AUTH_CLIENT_REDIRECT_URI,
+  scope: process.env.IMJS_AUTH_CLIENT_SCOPES ?? "",
```
Comment (aruniverse) [inline packages/templates/cra-template-web-viewer/template/src/index.tsx:32]: do you need the fallback to empty strings?

dont the above if statements check if its defined?

**Review by aruniverse — APPROVED** [review]

File: packages/templates/cra-template-web-viewer/template/src/index.tsx (hunk 8 above)
Comment (Jake-Screen) [reply packages/templates/cra-template-web-viewer/template/src/index.tsx]: removed fallback, they are checked earlier
